apps/ledger-live-desktop/src/renderer/components/VaultSignerTransport.tsx

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import React, { useEffect } from "react";
+import { registerTransportModule } from "@ledgerhq/live-common/hw/index";
+import { retry } from "@ledgerhq/live-common/promise";
+import { VaultTransport } from "@ledgerhq/hw-transport-http";
+import { useSelector } from "react-redux";
+import { setDeviceMode, currentMode } from "@ledgerhq/live-common/hw/actions/app";
+import { vaultSignerSelector } from "~/renderer/reducers/settings";
+
+const originalDeviceMode = currentMode;
+
+/*
+This component "listens" (with useEffect()) for changes in settings.vaultSigner and update
+the vault-transport in the global modules object.
+As this transport needs to be in sync with settings we couldn't register it
+on the internal process.
+
+To avoid cluttering the transport modules global object, we have updated it
+to not blindly push in the array but to update it if it's already in the array
+
+I'ts also important to check for the id of the transport in the open() method because
+there is also IPC transport registered.
+*/
+
+const VaultSignerTransport = () => {
+  const { token, enabled, host, workspace } = useSelector(vaultSignerSelector);
+  useEffect(() => {
+    if (enabled && currentMode !== "polling") {
+      setDeviceMode("polling");
+    } else if (!enabled) {
+      setDeviceMode(originalDeviceMode);
+    }
+    registerTransportModule({
```

Comment:
`registerTransportModule` was not yet designed to be dynamically used. it probably works the first time, but the problem is that we don't have a way to "unregister" which will end up being registering multiple time each time the effect runs and could create race conditions.

I would suggest:
- to move `registerTransportModule` as a module definition level again, next to the others registerTransportModule (like the one we do in the live common setup logic). => it shouldn't be a problem as you will filter on the id
- as you need some extra information (host, token, workspace) in that transport, you may also need to have it part of the id. for instance `vault-transport?host=foo&token=bar&workspace=42`

Reply:
The problem is I need the transport to be registered each time `host/token/workspace` changes, that's why it's inside the `useEffect()`

Reply:
but then it could work if it's part of the id?

Reply:
I struggle to see what you mean tbh, if I register the vault transport [here](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/live-common-setup.ts#L24) how can make it re-register each time `host/token/workspace` change?

Reply:
example https://github.com/LedgerHQ/ledger-live/blob/main/libs/ledger-live-common/src/load/speculos.ts#L471-L496